blockly-core/core/ui/block.js

```diff
@@ -1454,6 +1472,69 @@ Blockly.Block.prototype.setMovable = function(movable) {
 };
 
 /**
+ * Get whether this block represents a toolbox block that can only be
+ * used a limited number of times
+ * @return {boolean}
+ */
+Blockly.Block.prototype.hasLimit = function() {
+  return (this.isInFlyout || Blockly.editBlocks) && this.limit_ !== undefined;
+};
+
+/**
+ * If this block is a toolbox block, mark it as being able to "produce"
+ * a limited number of workspace blocks
+ * @param {number} limit
+ */
+Blockly.Block.prototype.setLimit = function(limit) {
+  limit = parseInt(limit);
+  if (isNaN(limit)) {
+    limit = undefined;
+  }
+  // only allow limits for toolbox blocks or when editing blocks (so
+  // that we can edit toolbox blocks)
+  if (this.isInFlyout || Blockly.editBlocks) {
```

Comment:
Ah makes sense now. Could maybe extract both usages of this into a `block.shouldRenderLimit()` helper instead, and move the above comment into there